miden-crypto/src/ecdh/k256.rs

```diff
@@ -0,0 +1,171 @@
+    /// to extract entropy from the shared secret.
+    ///
+    /// This basically converts a shared secret into uniformly random values that are appropriate
+    /// for use as key material.
+    pub fn extract<D>(&self, salt: Option<&[u8]>) -> Hkdf<Sha256, SimpleHmac<Sha256>> {
+        self.inner.extract(salt)
+    }
+}
+
+/// Ephemeral secret key for ECDH key agreement over secp256k1 curve.
+pub struct EphemeralSecretKey {
+    inner: k256::ecdh::EphemeralSecret,
+}
+
+impl EphemeralSecretKey {
+    /// Generates a new random ephemeral secret key using the OS random number generator.
+    #[cfg(feature = "std")]
+    #[allow(clippy::new_without_default)]
+    pub fn new() -> Self {
+        let mut rng = rand::rng();
+
+        Self::with_rng(&mut rng)
+    }
+
+    /// Generates a new ephemeral secret key using the provided random number generator.
+    pub fn with_rng<R: CryptoRng + RngCore>(rng: &mut R) -> Self {
+        use k256::elliptic_curve::rand_core::SeedableRng;
+
+        let mut seed = [0_u8; 32];
+        rand::RngCore::fill_bytes(rng, &mut seed);
+
+        let mut rng = rand_hc::Hc128Rng::from_seed(seed);
+
+        let sk_e = k256::ecdh::EphemeralSecret::random(&mut rng);
+        Self { inner: sk_e }
+    }
+
+    /// Gets the corresponding ephemeral public key for this ephemeral secret key.
+    pub fn public_key(&self) -> EphemeralPublicKey {
+        let pk = self.inner.public_key();
+        EphemeralPublicKey { inner: pk }
+    }
+
+    /// Computes a Diffie-Hellman shared secret from an ephemeral secret key and the (static) public
+    /// key of the other party.
+    pub fn diffie_hellman(&self, pk_other: PublicKey) -> SharedSecret {
+        let shared_secret_inner = self.inner.diffie_hellman(&pk_other.inner.into());
+
+        SharedSecret { inner: shared_secret_inner }
+    }
+}
+
+/// Ephemeral public key for ECDH key agreement over secp256k1 curve.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct EphemeralPublicKey {
+    pub(crate) inner: k256::PublicKey,
+}
+
+impl EphemeralPublicKey {
+    /// Returns a reference to this ephemeral public key as an elliptic curve point in affine
+    /// coordinates.
+    pub fn as_affine(&self) -> &AffinePoint {
+        self.inner.as_affine()
+    }
+}
+
+impl Serializable for EphemeralPublicKey {
+    fn write_into<W: ByteWriter>(&self, target: &mut W) {
+        // Compressed format
+        let encoded = self.inner.to_encoded_point(true);
+
+        target.write_bytes(encoded.as_bytes());
+    }
+}
+
+impl Deserializable for EphemeralPublicKey {
+    fn read_from<R: ByteReader>(source: &mut R) -> Result<Self, DeserializationError> {
+        let bytes: [u8; PUBLIC_KEY_BYTES] = source.read_array()?;
+
+        let inner = k256::PublicKey::from_sec1_bytes(&bytes)
+            .map_err(|_| DeserializationError::InvalidValue("Invalid public key".to_string()))?;
+
+        Ok(Self { inner })
+    }
+}
+
+#[cfg(test)]
```

Comment:
nit: let's add a separator above this line.

Reply:
Done